File: cavalo.py
import peca


class Cavalo(peca.Peca):
    def __init__(self, posicao_origem_x, posicao_origem_y, cor):
        super().__init__(posicao_origem_x,posicao_origem_y, cor)
        

    def validar(self, posicao_x, posicao_y):
        if super().validar(posicao_x, posicao_y):
            if (posicao_x == self._posicao_atual_x+2) or (posicao_x == self._posicao_atual_x-2):
                if (posicao_y == self._posicao_atual_y+1) or (posicao_y == self._posicao_atual_y-1):
                    return True
            elif (posicao_y == self._posicao_atual_y+2) or (posicao_y == self._posicao_atual_y-2):
                if (posicao_x == self._posicao_atual_x+1) or (posicao_x == self._posicao_atual_x-1):
                    return True
            return False
                
    def movimentar(self, destino_x, destino_y):
        if self.validar(destino_x, destino_y):
            self._posicao_origem_x = self._posicao_atual_x
            self._posicao_origem_y = self._posicao_atual_y
            self._posicao_atual_x = destino_x
            self._posicao_atual_y = destino_y
            return True
        return False

File: Cavalo.java
package xadrez_jv;

public class Cavalo extends Peca{
	
	Cavalo(int posicaoOrigemX, int posicaoOrigemY, int cor)
	{
		super(posicaoOrigemX, posicaoOrigemY, cor);
	}
	
	@Override 
	public boolean validar(int posicaoX, int posicaoY)
	{
		if (super.validar(posicaoX, posicaoY)) {
			if (posicaoX == this.posicaoAtualX + 2 || posicaoX == this.posicaoAtualX - 2)
			{
				if (posicaoY == this.posicaoAtualY + 1 || posicaoY == posicaoAtualY - 1)
					return true;
			}else if(posicaoY == this.posicaoAtualY + 2 || posicaoY == posicaoAtualY - 2){
				if (posicaoX == this.posicaoAtualX + 1 || posicaoX == this.posicaoAtualX - 1)
					return true;
			}
		}
		return false;			
	}
	
	@Override 
	public boolean movimentar(int destinoX, int destinoY)
	{
		if (validar(destinoX, destinoY))
		{
			this.posicaoOrigemX = this.posicaoAtualX;
			this.posicaoOrigemY = this.posicaoAtualY;
			this.posicaoAtualX = destinoX;
			this.posicaoAtualY = destinoY;
			return true;
		}
		return false;
	}
}
